### ISSUE TITLE
Build Failure - Could not find llvm-config

### ISSUE THREAD

**Issue by carbotaniuman** [twistedfall/opencv-rust#136]
Before reporting an issue please first check the
[troubleshooting guide](https://github.com/twistedfall/opencv-rust/blob/master/README.md#troubleshooting). If
the issue you're encountering is not solved thereby please state the following in your bugreport:
1. Windows 10 2004
2. vcpkg - also installed llvm per readme.
3. opencv 4.1.1
4. Attach the full output of the following command from your project directory:
```
PS C:\Users\quydx\Documents\Code\vision_manager> cargo build -vv
       Fresh autocfg v0.1.6
       Fresh rand_core v0.4.2
       Fresh unicode-xid v0.2.0
       Fresh glob v0.3.0
       Fresh lazy_static v1.4.0
       Fresh regex-syntax v0.6.18
       Fresh semver-parser v0.7.0
       Fresh proc-macro-hack v0.5.15
       Fresh maplit v1.0.2
       Fresh percent-encoding v2.1.0
       Fresh once_cell v1.4.0
       Fresh dunce v1.0.1
       Fresh autocfg v1.0.0
       Fresh jobserver v0.1.21
       Fresh pkg-config v0.3.17
       Fresh shlex v0.1.1
       Fresh vcpkg v0.2.9
       Fresh json v0.11.15
       Fresh rand_core v0.3.1
       Fresh thread_local v1.0.1
       Fresh cc v1.0.54
       Fresh semver v0.10.0
       Fresh libc v0.2.62
       Fresh proc-macro2 v1.0.1
       Fresh memchr v2.3.3
       Fresh winapi v0.3.8
       Fresh rand_hc v0.1.0
       Fresh rand_isaac v0.1.1
       Fresh rand_xorshift v0.1.1
warning: couldn't execute `llvm-config --prefix` (error: The system cannot find the file specified. (os error 2))
warning: set the LLVM_CONFIG_PATH environment variable to a valid `llvm-config` executable
warning: couldn't execute `llvm-config --includedir` (error: The system cannot find the file specified. (os error 2))
warning: set the LLVM_CONFIG_PATH environment variable to a valid `llvm-config` executable
       Fresh quote v1.0.2
       Fresh aho-corasick v0.7.10
       Fresh clang-sys v0.28.1
       Fresh rand_jitter v0.1.4
       Fresh rand_os v0.1.3
       Fresh rand_chacha v0.1.1
       Fresh rand_pcg v0.1.2
       Fresh num-traits v0.2.11
       Fresh syn v1.0.5
       Fresh clang v0.23.0
       Fresh regex v1.3.9
       Fresh rand v0.6.5
       Fresh opencv-binding-generator v0.13.0
       Fresh paste-impl v0.1.12
       Fresh vision_traits_derive v0.1.0 (C:\Users\quydx\Documents\Code\vision_traits\vision_traits_derive)
       Fresh paste v0.1.12
       Fresh uuid v0.7.4
       Fresh vision_traits v0.1.0 (C:\Users\quydx\Documents\Code\vision_traits)
   Compiling opencv v0.40.0
       Fresh modules v0.1.0 (C:\Users\quydx\Documents\Code\vision_manager\modules)
     Running `C:\Users\quydx\Documents\Code\vision_manager\target\debug\build\opencv-3c025c41909f3eae\build-script-build`
error: failed to run custom build command for `opencv v0.40.0`

Caused by:
  process didn't exit successfully: `C:\Users\quydx\Documents\Code\vision_manager\target\debug\build\opencv-3c025c41909f3eae\build-script-build` (exit code: 0xc000007b)
```
I checked the LLVM_CONFIG_PATH env var and it outputted the executable, so I'm not sure what the cause is.

**Comment by carbotaniuman** [comment]
I'm also pretty sure that's not the cause of this failure, but I don't see any other error messages.

**Comment by twistedfall** [comment]
You are most probably right, this is just a warning by clang crate. But the log you’re showing is not full for some reason. Can you please try running `cargo clean` before `cargo build -vv`?

**Comment by carbotaniuman** [comment]
The first issue is because vcpkg installs x86 instead of x64 by default for some stupid reason.

I got another error.

```
    Running `set CARGO=\\?\C:\Users\quydx\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin\cargo.exe&& set CARGO_MANIFEST_DIR=C:\Users\quydx\.cargo\registry\src\github.com-1ecc6299db9ec823\opencv-0.40.0&& set CARGO_PKG_AUTHORS="Mathieu Poumeyrol <kali@zoy.org>:Pro <twisted.fall@gmail.com>"&& set CARGO_PKG_DESCRIPTION="Rust bindings for OpenCV"&& set CARGO_PKG_HOMEPAGE=&& set CARGO_PKG_NAME=opencv&& set CARGO_PKG_REPOSITORY=https://github.com/twistedfall/opencv-rust&& set CARGO_PKG_VERSION=0.40.0&& set CARGO_PKG_VERSION_MAJOR=0&& set CARGO_PKG_VERSION_MINOR=40&& set CARGO_PKG_VERSION_PATCH=0&& set CARGO_PKG_VERSION_PRE=&& set OUT_DIR=C:\Users\quydx\Documents\Code\vision_manager\target\debug\build\opencv-d78ee65b4140ba70\out&& set PATH="C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps;C:\Users\quydx\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;C:\Users\quydx\.cargo\bin;C:\Users\quydx\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;C:\Program Files\AdoptOpenJDK\jdk-8.0.242.08-hotspot\bin;C:\Program Files\AdoptOpenJDK\jdk-13.0.1.9-hotspot\bin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\PuTTY\;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\Git\cmd;C:\Program Files\CMake\bin;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Yarn\bin\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Program Files\dotnet\;C:\Users\Public\wpilib\2020\roborio\bin;C:\Users\Public\wpilib\2020\frccode;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Go\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Users\quydx\Downloads\vcpkg\installed\x64-windows\bin;C:\Users\quydx\Downloads\vcpkg\installed\x64-windows\debug\bin;C:\lscc\iCEcube2.2017.08\Aldec\Active-HDL\BIN;C:\Users\quydx\AppData\Local\Programs\Python\Python37\Scripts\;C:\Users\quydx\AppData\Local\Programs\Python\Python37\;C:\Users\quydx\.cargo\bin;C:\Users\quydx\AppData\Local\Microsoft\WindowsApps;C:\Users\quydx\AppData\Local\GitHubDesktop\bin;C:\Users\quydx\AppData\Local\Yarn\bin;C:\Users\quydx\AppData\Roaming\npm;C:\Users\Public\wpilib\2020\roborio\bin;C:\Users\Public\wpilib\2020\frccode;C:\Users\quydx\AppData\Local\Programs\radare2\bin;C:\Users\quydx\AppData\Local\Microsoft\WindowsApps;C:\Users\quydx\go\bin;"&& rustc --crate-name opencv --edition=2018 C:\Users\quydx\.cargo\registry\src\github.com-1ecc6299db9ec823\opencv-0.40.0\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -Cembed-bitcode=no -C debuginfo=2 --cfg "feature=\"binding-generator\"" --cfg "feature=\"buildtime-bindgen\"" --cfg "feature=\"clang\"" --cfg "feature=\"default\"" --cfg "feature=\"jobserver\"" --cfg "feature=\"opencv-4\"" -C metadata=643f2de404752001 -C extra-filename=-643f2de404752001 --out-dir C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps -L dependency=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps --extern libc=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps\liblibc-8b5bd677cf258683.rmeta --extern num_traits=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps\libnum_traits-f7dc2e8b0b4e34ed.rmeta --extern once_cell=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps\libonce_cell-e14e09b3ff227454.rmeta --cap-lints warn -L native=C:\Users\quydx\Documents\Code\vision_manager\target\debug\build\opencv-d78ee65b4140ba70\out -L native= -L native=C:/Users/quydx/Downloads/vcpkg/installed/x64-windows/debug/lib -l static=ocvrs -l opencv_dnnd -l opencv_highguid -l opencv_mld -l opencv_objdetectd -l opencv_photod -l opencv_stitchingd -l opencv_videod -l opencv_videoiod -l opencv_imgcodecsd -l opencv_calib3dd -l opencv_features2dd -l opencv_flannd -l opencv_imgprocd -l opencv_cored -l `
error: empty search path given via `-L`

error: could not compile `opencv`.

Caused by:
  process didn't exit successfully: `set CARGO=\\?\C:\Users\quydx\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin\cargo.exe&& set CARGO_MANIFEST_DIR=C:\Users\quydx\.cargo\registry\src\github.com-1ecc6299db9ec823\opencv-0.40.0&& set CARGO_PKG_AUTHORS="Mathieu Poumeyrol <kali@zoy.org>:Pro <twisted.fall@gmail.com>"&& set CARGO_PKG_DESCRIPTION="Rust bindings for OpenCV"&& set CARGO_PKG_HOMEPAGE=&& set CARGO_PKG_NAME=opencv&& set CARGO_PKG_REPOSITORY=https://github.com/twistedfall/opencv-rust&& set CARGO_PKG_VERSION=0.40.0&& set CARGO_PKG_VERSION_MAJOR=0&& set CARGO_PKG_VERSION_MINOR=40&& set CARGO_PKG_VERSION_PATCH=0&& set CARGO_PKG_VERSION_PRE=&& set OUT_DIR=C:\Users\quydx\Documents\Code\vision_manager\target\debug\build\opencv-d78ee65b4140ba70\out&& set PATH="C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps;C:\Users\quydx\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;C:\Users\quydx\.cargo\bin;C:\Users\quydx\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;C:\Program Files\AdoptOpenJDK\jdk-8.0.242.08-hotspot\bin;C:\Program Files\AdoptOpenJDK\jdk-13.0.1.9-hotspot\bin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\PuTTY\;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\Git\cmd;C:\Program Files\CMake\bin;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Yarn\bin\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Program Files\dotnet\;C:\Users\Public\wpilib\2020\roborio\bin;C:\Users\Public\wpilib\2020\frccode;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Go\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Users\quydx\Downloads\vcpkg\installed\x64-windows\bin;C:\Users\quydx\Downloads\vcpkg\installed\x64-windows\debug\bin;C:\lscc\iCEcube2.2017.08\Aldec\Active-HDL\BIN;C:\Users\quydx\AppData\Local\Programs\Python\Python37\Scripts\;C:\Users\quydx\AppData\Local\Programs\Python\Python37\;C:\Users\quydx\.cargo\bin;C:\Users\quydx\AppData\Local\Microsoft\WindowsApps;C:\Users\quydx\AppData\Local\GitHubDesktop\bin;C:\Users\quydx\AppData\Local\Yarn\bin;C:\Users\quydx\AppData\Roaming\npm;C:\Users\Public\wpilib\2020\roborio\bin;C:\Users\Public\wpilib\2020\frccode;C:\Users\quydx\AppData\Local\Programs\radare2\bin;C:\Users\quydx\AppData\Local\Microsoft\WindowsApps;C:\Users\quydx\go\bin;"&& rustc --crate-name opencv --edition=2018 C:\Users\quydx\.cargo\registry\src\github.com-1ecc6299db9ec823\opencv-0.40.0\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -Cembed-bitcode=no -C debuginfo=2 --cfg "feature=\"binding-generator\"" --cfg "feature=\"buildtime-bindgen\"" --cfg "feature=\"clang\"" --cfg "feature=\"default\"" --cfg "feature=\"jobserver\"" --cfg "feature=\"opencv-4\"" -C metadata=643f2de404752001 -C extra-filename=-643f2de404752001 --out-dir C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps -L dependency=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps --extern libc=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps\liblibc-8b5bd677cf258683.rmeta --extern num_traits=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps\libnum_traits-f7dc2e8b0b4e34ed.rmeta --extern once_cell=C:\Users\quydx\Documents\Code\vision_manager\target\debug\deps\libonce_cell-e14e09b3ff227454.rmeta --cap-lints warn -L native=C:\Users\quydx\Documents\Code\vision_manager\target\debug\build\opencv-d78ee65b4140ba70\out -L native= -L native=C:/Users/quydx/Downloads/vcpkg/installed/x64-windows/debug/lib -l static=ocvrs -l opencv_dnnd -l opencv_highguid -l opencv_mld 
-l opencv_objdetectd -l opencv_photod -l opencv_stitchingd -l opencv_videod -l opencv_videoiod -l opencv_imgcodecsd -l opencv_calib3dd -l opencv_features2dd -l opencv_flannd -l opencv_imgprocd -l opencv_cored -l ` (exit code: 1)
```

I've copied only a portion because all of the logs overflowed my console.

**Comment by twistedfall** [comment]
I'd still like to see the full build log. The error "error: empty search path given via `-L`" indicates that OpenCV library discovery went not so smooth and the full build log will give us the information as to why.

**Comment by carbotaniuman** [comment]
```
=== Crate version: Some("0.40.0")
=== Environment configuration:
===   OPENCV_HEADER_DIR = None
===   OPENCV_PACKAGE_NAME = None
===   OPENCV_PKGCONFIG_NAME = None
===   OPENCV_CMAKE_NAME = None
===   OPENCV_CMAKE_BIN = None
===   OPENCV_VCPKG_NAME = None
===   OPENCV_LINK_LIBS = None
===   OPENCV_LINK_PATHS = None
===   OPENCV_INCLUDE_PATHS = None
===   OPENCV_DISABLE_PROBES = None
===   OPENCV_CLANG_STDLIB_PATH = None
===   CMAKE_PREFIX_PATH = None
===   OpenCV_DIR = None
===   PKG_CONFIG_PATH = None
===   VCPKG_ROOT = None
=== Enabled features:
===   BINDING_GENERATOR
===   BUILDTIME_BINDGEN
===   CLANG
===   DEFAULT
===   JOBSERVER
===   OPENCV_4
=== Probing OpenCV library using pkg_config
=== Can't probe using pkg_config, continuing with other methods, error: Failed to run `"pkg-config" "--libs" "--cflags" "opencv4"`: The system cannot find the file specified. (os error 2)
=== Probing OpenCV library using cmake
=== Unexpected cmake compile argument found: 
=== OpenCV library configuration: Library {
    include_paths: [
        "C:/Users/quydx/Downloads/vcpkg/installed/x64-windows/include",
    ],
    version: "4.1.1",
    cargo_metadata: [
        "cargo:rustc-link-lib=opencv_dnnd",
        "cargo:rustc-link-lib=opencv_highguid",
        "cargo:rustc-link-lib=opencv_mld",
        "cargo:rustc-link-lib=opencv_objdetectd",
        "cargo:rustc-link-lib=opencv_photod",
        "cargo:rustc-link-lib=opencv_stitchingd",
        "cargo:rustc-link-lib=opencv_videod",
        "cargo:rustc-link-lib=opencv_videoiod",
        "cargo:rustc-link-lib=opencv_imgcodecsd",
        "cargo:rustc-link-lib=opencv_calib3dd",
        "cargo:rustc-link-lib=opencv_features2dd",
        "cargo:rustc-link-lib=opencv_flannd",
        "cargo:rustc-link-lib=opencv_imgprocd",
        "cargo:rustc-link-lib=opencv_cored",
        "cargo:rustc-link-lib=\r",
        "cargo:rustc-link-search=native=",
        "cargo:rustc-link-search=native=C:/Users/quydx/Downloads/vcpkg/installed/x64-windows/debug/lib",
    ],
}
=== Found OpenCV library version: 4.1.1 in headers located at: C:/Users/quydx/Downloads/vcpkg/installed/x64-windows/include
=== Using OpenCV headers from: C:/Users/quydx/Downloads/vcpkg/installed/x64-windows/include\opencv2
=== Generating code in: C:\Users\quydx\Documents\Code\vision_manager\target\debug\build\opencv-3937b6e0052cdea8\out
=== Placing generated bindings into: C:\Users\quydx\.cargo\registry\src\github.com-1ecc6299db9ec823\opencv-0.40.0\bindings\rust\opencv_4
=== Clang command line args: [
    "-isystemC:\\Users\\quydx\\Downloads\\vcpkg\\installed\\x64-windows\\lib\\clang\\10.0.0\\include",
    "-isystemC:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.22.27905\\include",
    "-isystemC:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.22.27905\\atlmfc\\include",
    "-isystemC:\\Program Files (x86)\\Windows Kits\\10\\Include\\10.0.18362.0\\ucrt",
    "-isystemC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared",
    "-isystemC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um",
    "-isystemC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt",
    "-IC:\\Users\\quydx\\Downloads\\vcpkg\\installed\\x64-windows\\include",
    "-IC:\\Users\\quydx\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\opencv-0.40.0\\src_cpp",
    "-DOCVRS_PARSING_HEADERS",
    "-includeocvrs_resolve_types.hpp",
]
```

**Comment by twistedfall** [comment]
Yeah, that's bad:
```
...
        "cargo:rustc-link-lib=\r",
        "cargo:rustc-link-search=native=",
...
```

You should be able to work it around with `OPENCV_DISABLE_PROBES` environment variable set to "cmake". I'll see if I can reproduce it locally.

**Comment by twistedfall** [comment]
Can you please check the current master? It should be fixed and not require any workarounds with OPENCV_DISABLE_PROBES. If it works ok for you I'll do the point release

**Comment by carbotaniuman** [comment]
Yay! It works. Thanks a lot.

**Comment by twistedfall** [comment]
Fixed in 0.40.1
